sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -26,7 +26,7 @@ import org.apache.spark.sql.catalyst.dsl.expressions._
 import org.apache.spark.sql.catalyst.expressions._
 import org.apache.spark.sql.types._
 
-case class Average(child: Expression) extends AlgebraicAggregate {
+case class Average(child: Expression) extends ExpressionAggregateFunction {
```

Comment:
@rxin and I think that algebraic aggregate is potentially confusing because the [Jim Gray Data Cube paper](http://arxiv.org/pdf/cs/0701155.pdf) already defined the term "algebraic aggregate" to mean something slightly differently than how we're using it here. Technically, an algebraic aggregate (in the Gray terminology) could be implemented via an imperative aggregation function.